src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -105,32 +108,124 @@ public static void runQueryBenchmarks(List<QueryBenchmark> queryBenchmarks, Stri
 		    for (int threads = benchmark.minThreads; threads <= benchmark.maxThreads; threads++) {
 		        QueryGenerator queryGenerator = new QueryGenerator(benchmark);
 		        HttpSolrClient client = new HttpSolrClient.Builder(baseUrl).build();
+		        String collection = collectionNameOverride==null? benchmark.collection: collectionNameOverride;
 		        ControlledExecutor controlledExecutor = new ControlledExecutor(threads,
 		                benchmark.duration,
 		                benchmark.rpm,
 		                benchmark.totalCount,
 		                benchmark.warmCount,
-		                getQuerySupplier(queryGenerator, client, collectionNameOverride==null? benchmark.collection: collectionNameOverride));
+		                getQuerySupplier(queryGenerator, client, collection));
 		        long start = System.currentTimeMillis();
+						List<SolrBenchQuery> generatedQueries;
 		        try {
-		            controlledExecutor.run();
+		            generatedQueries = controlledExecutor.run();
 		        } finally {
 		            client.close();
 		        }
 
+	        	int success = 0, failure = 0;
+
+		        if (StressMain.generateValidations) {
+		        	int totalDocsIndexed = getTotalDocsIndexed(baseUrl, collection);
+		        			
+		        	FileWriter outFile = new FileWriter("suites/validations-"+benchmark.queryFile+"-docs-"+totalDocsIndexed+"-queries-"+benchmark.totalCount+".tsv");
+			        for (SolrBenchQuery sbq: generatedQueries) {
+			        	int numFound = getNumFoundFromSolrQueryResponse(sbq.response);
+			        	Map<String, Object> facets = getFacetsFromSolrQueryResponse(sbq.response);
+			        	outFile.write(sbq.queryString.replace('\n', ' ').replace('\r', ' ') +
+			        			"\t" + numFound + "\t" + new ObjectMapper().writeValueAsString(facets).replace('\n', ' ') + "\n");
+			        }
+			        outFile.close();
+		        } else if (StressMain.validate && benchmark.validationFile != null && new File("suites/" + benchmark.validationFile).exists()) {
```

Comment:
🧠 storming a little bit here. Wondering if we can make this more abstract and provide at least 2 implementations:
1. The validator that's based on an previously generated input file - as implemented here
2. The validator that's based on more static criteria - might be as simple as just verify the doc count as > 0

I do think that the current validation proposal (with a file) would be the useful in most cases, however, when I think about the FS test workflow, we might not really want to further complicate the procedure of input doc/query update. 

For the file validation flow, every time we update query/index, we will need to run it once as K8s but probably with a different task (generate result), capture the validation file, then update the corresponding image with the new validation file etc. 

It's not super complicated, but it's not trivial neither. I think perhaps a middle ground is that : as far as the result return something, we assume they are legit results. 

The main goal of solr-bench is to confirm the performance change due to code changes, the validation imho is to flag any obvious anomaly (such as not returning anything at all) that might impact the accuracy of benchmarking. While it's nice if the validation works more accurately (as suggested by the validation file approach), the added burden to maintain such file might not be that appealing.

Any thoughts? @justinrsweeney @hiteshk25 @chatman 😊

Reply:
>     The validator that's based on a previously generated input file - as implemented here
>    The validator that's based on more static criteria - might be as simple as just verify the doc count as > 0

That makes sense. We can have validations as a complex JSON object in the configuration, and specify either we blindly compare against a file (your point 1), or some special conditions (e.g. results > 0) etc.